### ISSUE TITLE
Can't get code to work :(

### ISSUE THREAD

**Issue by richardhughes260** [bren1818/TCPLightingWebInterface#3]
It may be me, not you.

Is the only thing I need to get started is the following in the include.php:

IP of controller
token captured by some external method
ip of the host of this code

I get the message that I do not have the token filled in, it is, on line 29.

**Comment by bren1818** [comment]
You need to set the ip to the ip of the tcp bridge, then push the sync on
the bridge and go to where you run the app. It should give you a token,
which you copy and then paste into the include file.

I'll have a look tonight and push an update if need be.

Good luck!

**Comment by richardhughes260** [comment]
I loaded fresh code, added the ip of the lighting controller in include.php & rand index.php after pushing the sync button. I got the same message, that It did not get a token.

Is there other code that needs to be run to get the token?

**Comment by bren1818** [comment]
Hmm I'll have to look into it tonight,  default settings should work after
those edits.  Could be firewall issue.  What OS are you using?

Thanks

**Comment by richardhughes260** [comment]
I am running it on a Orange Pi PC, let me check permissions...

**Comment by richardhughes260** [comment]
I did a chmod 777 \* in the dir where in the index.php is. Still no success. This is all on the LAN, so no FW issues, if it is not phoning home.

**Comment by bren1818** [comment]
Hmm I'll have a look tonight when I'm home. Could you try setting line 17
in includes to 0,  so 'use token file' is false?

Many thanks

**Comment by bren1818** [comment]
Iam unable to replicate the issue you;re encountering, however try the following:
in the includes.php file
set:
line 8: LIGHTING_URL to the address of the bridge
line 11: LOCAL_URL to: "127.0.0.1" (try this, localhost should work too, depends on how Orange Pi runs the webserver)
Line 17: USE_TOKEN_FILE to 0

then run it, you should see:

![image](https://cloud.githubusercontent.com/assets/2146847/19010359/ca2eb590-874c-11e6-81cc-75e51d70252c.png)

Hit the sync button and re-fresh

It should generate your token, you want the highlighted part:

![image](https://cloud.githubusercontent.com/assets/2146847/19010369/ffc8fe54-874c-11e6-911c-f1e0d73e8743.png)

Put that as the token in line 29

![image](https://cloud.githubusercontent.com/assets/2146847/19010385/207d5686-874d-11e6-99d2-fc9bbb66877e.png)

Then you should be good.. Your config should look something like this: (places you need to adjust are highlighted)

![image](https://cloud.githubusercontent.com/assets/2146847/19010402/629d2f82-874d-11e6-8acc-dae3cfd3c16a.png)

**Comment by richardhughes260** [comment]
Hi Bren, changing Line 17: USE_TOKEN_FILE to 0 made the token warning go away. Nothing else made any changes. 

It may be the OS. I'll have to deploy it on a mac, pc or linux distro.

I was able to get a token using a python app. I tried pasting it in & no better results.

**Comment by sdhip** [comment]
Hi Bren

Great work -- well done (and to all those whose work you built on).

I tried with the wireless tcp bridge and had the errors that others mention above, i.e. no token returned, but then switched to a wired one I had spare from buying some bulbs on eBay and it all worked just fine.

Maybe the problems folk are reporting are down to wireless tcp bridges?

All the best
Steve

**Comment by bren1818** [comment]
Very interesting hypothesis, that may be the issue as I  could not
replicate it. I was unaware there even was a wireless version of the hub.
Perhaps if  I pick one up, I'll be able to find a work around. Thank you
for posting that info.

Cheers, Bren

**Comment by bren1818** [comment]
For those of you using Windows 10, something I thought of (as I had issues connecting wireless to my Raspberry Pi - and this may be completely unrelated) is the following:

Click on your wireless network

![image](https://cloud.githubusercontent.com/assets/2146847/19214180/3c49f992-8d4b-11e6-9afa-03fd39a5f42f.png)

Once connected click properties

![image](https://cloud.githubusercontent.com/assets/2146847/19214184/5c0c9dd4-8d4b-11e6-8b74-6a45eac94dd7.png)

Under make this PC discoverable, click yes.

![image](https://cloud.githubusercontent.com/assets/2146847/19214187/811089ce-8d4b-11e6-8b40-8c7afb1c78b1.png)

I'm not sure if this will help, but I know after one of the Windows 10 updates, I couldn't connect to other items on my network, like smb on my Raspberry Pi, but this fixed it.

Try it out, let me know.

Thanks!

@richardhughes260 @sdhip 

,

**Comment by sdhip** [comment]
Hi Bren

Wireless gateway is in the bulb packs with a white iPhone on the front:
![s-l1600](https://cloud.githubusercontent.com/assets/22703825/19220470/7b60b4d8-8e25-11e6-8ed2-9b1c1b1b2920.jpg)

Let me have your address Bren if you'd like me to send you one!

I'm running your code on a Mac (with standard Apache & PHP setup -- see e.g. https://coolestguidesontheplanet.com/get-apache-mysql-php-and-phpmyadmin-working-on-macos-sierra/).

Cheers
Steve

**Comment by bren1818** [comment]
@sdhip Steve, that's very generous of you and would be awesome. I have not seen one of those packs before.  Perhaps that version was not available in Canada or atleast in the Home Depots.  I'm going to assume if the code did work with the wired version but not the wireless bridge perhaps it is either communicating on a different port or there is a issue in between. Out of curiosity is your Mac connected wired to your router or also connected wirelessly to it (not that it should make a difference) 

I cannot imagine there is different api commands for the wireless version,  I have decompiled the Android source so I'll be able to look at that a bit more scrutinizingly this week. I keep intending to add more features to this project like bulb discovery etc but just haven't had the time lately. 

I'll send you my address details this week, I'm just on the road to visit family.  

Thanks again! 

Bren

**Comment by bren1818** [comment]
@sdhip, @richardhughes260 - I was able to replicate the issue using the "official" android app on a wired bridge. (I think) - At my home I have my primary wireless network, and I have a wireless Bridg/Repeater Network too. My TCP bridge is connected to my primary network. When I tried to connect to my Bridged network, the application would not ever find the lights, it would fail, however, when I connected my phone to my primary network and ran the app, It did connect. This may be another thing to check - ensure the bridge and where you're controlling the lights from are all connected to the same hub / network.
